lib/Conversion/HandshakeToFIRRTL/HandshakeToFIRRTL.cpp

```diff
@@ -527,16 +530,19 @@ static FModuleOp createTopModuleOp(handshake::FuncOp funcOp, unsigned numClocks,
 // FIRRTL Sub-module Related Functions
 //===----------------------------------------------------------------------===//
 
-/// Check whether a submodule with the same name has been created elsewhere.
-/// Return the matched submodule if true, otherwise return nullptr.
-static FModuleOp checkSubModuleOp(FModuleOp topModuleOp, Operation *oldOp) {
-  for (auto &op : topModuleOp->getParentRegion()->front()) {
-    if (auto subModuleOp = dyn_cast<FModuleOp>(op)) {
-      if (getSubModuleName(oldOp) == subModuleOp.getName()) {
-        return subModuleOp;
-      }
-    }
+/// Check whether a submodule with the same name has been created elsewhere in
+/// the FIRRTL circt. Return the matched submodule if true, otherwise return
+/// nullptr.
+static FModuleOp checkSubModuleOp(CircuitOp circuitOp, Operation *oldOp) {
+  for (auto moduleOp : circuitOp.getOps<FModuleOp>()) {
```

Comment:
nit: could this just be a symbol lookup on the `circuitOp`? I don't think that will be much more efficient honestly, but seems like the code will be a tad simpler if possible.

Reply:
good suggestion, will make the change :+1: .